lib/ramble/ramble/context.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2022-2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from ramble.namespace import namespace
+import ramble.util.matrices
+
+import spack.util.spack_yaml as syaml
+
+
+class Context(object):
+    """Class to represent a context
+
+    This class contains variable definitions to store any individual context
+    (such as application, workload, or experiment) and logic to merge in
+    additional contexts by order of precedence."""
+
+    def __init__(self):
+        """Constructor for a Context
+
+        Create a Context object, which holds context attributes.
+        """
+        self.env_variables = []
+        self.variables = syaml.syaml_dict()
+        self.internals = {}
+        self.templates = None
+        self.chained_experiments = []
+        self.modifiers = []
+        self.context_names = None
```

Comment:
One other small nit, that I just noticed when you changed the name of the `from_dict` function.

Is there a reason for this to be plural? It seems like it should be singular instead. But, if there is a reason then we can leave it.